src/pam_hbac_dnparse.c

```diff
@@ -48,6 +48,36 @@ rdn_keyval_matches(LDAPAVA **rdn, const char *key, const char *val)
     return matches;
 }
 
+static bool
+dn_matches(LDAPAVA *dn, LDAPAVA *dn2)
+{
+    if (dn == NULL || dn2 == NULL) {
+        return false;
+    }
+
+    if (dn->la_attr.bv_len != dn2->la_attr.bv_len) {
+        return false;
+    }
+
+    if (dn->la_value.bv_len != dn2->la_value.bv_len) {
+        return false;
+    }
+
+    if (strncasecmp(dn->la_attr.bv_val,
+                    dn2->la_attr.bv_val,
+                    dn->la_attr.bv_len) != 0) {
```

Comment:
Can you add a test where the DN differs in RDN attribute name so that this branch is tested (cn=ipa,dc=test versus dc=ipa,dc=test should work).